File: 백준/Bronze/2139. 나는 너가 살아온 날을 알고 있다/나는 너가 살아온 날을 알고 있다.py
def is_leap_year(year):
    if year % 400 == 0:
        return True
    if year % 100 == 0:
        return False
    return year % 4 == 0


while True:
    d, m, y = map(int, input().split())
    if not d and not m and not y:
        break

    days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    if is_leap_year(y):
        days[1] += 1

    print(sum(days[:m-1]) + d)

File: 백준/Bronze/2139. 나는 너가 살아온 날을 알고 있다/나는 너가 살아온 날을 알고 있다.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        while (true) {
            int[] dmy = Arrays.stream(br.readLine().split(" "))
                    .mapToInt(Integer::parseInt)
                    .toArray();

            int d = dmy[0];
            int m = dmy[1];
            int y = dmy[2];

            if (d == 0 && m == 0 && y == 0)
                return;

            int[] days = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
            if (isLeapYear(y))
                days[1]++;

            System.out.println(Arrays.stream(days, 0, m - 1).sum() + d);
        }
    }

    public static boolean isLeapYear(int year) {
        if (year % 400 == 0)
            return true;
        if (year % 100 == 0)
            return false;
        return year % 4 == 0;
    }
}